connectors/bigtable/flink-examples-gcp-bigtable/src/main/java/com/google/flink/connector/gcp/bigtable/examples/WriteTableAPINested.java

```diff
@@ -0,0 +1,120 @@
+
+/**
+ * This is an example pipeline for Apache Flink that demonstrates writing data to Google Cloud
+ * Bigtable using the Bigtable connector, nested rows and the Table API.
+ *
+ * <p>The pipeline generates a stream of rows with a string key, a string column, and an integer
+ * column using the "datagen" connector. It then writes this data to Bigtable using the Table API
+ * and the Bigtable connector.
+ *
+ * <p>To run this example, you need to pass the argument {@code --columnFamily} or have an existing
+ * column family named {@code flink}.
+ *
+ * <p>You can run this example by passing the following command line arguments:
+ *
+ * <pre>
+ *   --instance &lt;bigtable instance id&gt; \
+ *   --project &lt;gcp project id&gt; \
+ *   --table &lt;bigtable table id&gt; \
+ *   --rate &lt;number of rows to generate per second&gt; \
+ *   --jobName &lt;job name&gt;
+ * </pre>
+ */
+public class WriteTableAPINested {
+    public static void main(String[] args) throws Exception {
+
+        final ParameterTool parameterTool = ParameterTool.fromArgs(args);
+        String instance = parameterTool.get("instance");
+        String project = parameterTool.get("project");
+        String table = parameterTool.get("table");
+        Long rate = parameterTool.getLong("rate", 100L);
+        String jobName = parameterTool.get("jobName", "Streaming Bigtable Write Table API Nested");
+
+        EnvironmentSettings settings = EnvironmentSettings.newInstance().inStreamingMode().build();
+
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.getConfig().setGlobalJobParameters(parameterTool);
+        env.enableCheckpointing(parameterTool.getInt("interval", 1000));
+
+        StreamTableEnvironment tableEnv = StreamTableEnvironment.create(env, settings);
+
+        Schema schemaDatagen =
+                Schema.newBuilder()
+                        .column("my-key", DataTypes.STRING().notNull())
+                        .column(
+                                "family1",
+                                DataTypes.ROW(
+                                        DataTypes.FIELD("stringColumn", DataTypes.STRING()),
+                                        DataTypes.FIELD("intColumn", DataTypes.INT())))
+                        .column(
+                                "family2",
+                                DataTypes.ROW(DataTypes.FIELD("stringColumn2", DataTypes.STRING())))
+                        .build();
+
+        Schema schemaBigtable =
+                Schema.newBuilder()
+                        .column("my-key", DataTypes.STRING().notNull())
+                        .column(
+                                "family1",
+                                DataTypes.ROW(
+                                        DataTypes.FIELD("stringColumn", DataTypes.STRING()),
+                                        DataTypes.FIELD("intColumn", DataTypes.INT())))
+                        .column(
```

Comment:
this is kind of unrelated to this PR but what happens if a user is using nested rows mode and doesnt specify a column family for a column that is not nested? does that question make sense...

Reply:
All of that is tested, bothn in Generic and RowData. For Table API, it uses RowData tests as it uses it as serializer:

https://github.com/google/flink-connector-gcp/blob/main/connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java#L147